catalyst-gateway/bin/src/service/api/cardano/rbac/registrations_get/chain_info.rs

```diff
@@ -30,76 +17,30 @@ pub struct ChainInfo {
 
 impl ChainInfo {
     /// Creates a new `ChainInfo` instance.
-    pub(crate) async fn new(
-        persistent_session: &CassandraSession, volatile_session: &CassandraSession,
-        catalyst_id: &CatalystId,
-    ) -> anyhow::Result<Option<Self>> {
-        let registrations =
-            last_registration_chain(persistent_session, volatile_session, catalyst_id).await?;
-
-        let network = Settings::cardano_network();
-
+    pub(crate) fn new(
+        persistent_chain: Option<RegistrationChain>, volatile_chain: Option<RegistrationChain>,
+    ) -> Option<Self> {
+        let mut chain = None;
         let mut last_persistent_txn = None;
         let mut last_volatile_txn = None;
         let mut last_persistent_slot = 0.into();
 
-        let res = build_reg_chain(
-            registrations.into_iter(),
-            network,
-            |is_persistent: bool, slot_no: Slot, chain: &RegistrationChain| {
-                if is_persistent {
-                    last_persistent_txn = Some(chain.current_tx_id_hash());
-                    last_persistent_slot = slot_no;
-                } else {
-                    last_volatile_txn = Some(chain.current_tx_id_hash());
-                }
-            },
-        )
-        .await?
-        .map(|chain| {
-            Self {
-                chain,
-                last_persistent_txn,
-                last_volatile_txn,
-                last_persistent_slot,
-            }
-        });
-        Ok(res)
+        if let Some(c) = persistent_chain {
+            last_persistent_txn = Some(c.current_tx_id_hash());
+            last_persistent_slot = c.current_point().slot_or_default();
+            chain = Some(c);
+        };
+        if let Some(c) = volatile_chain {
+            last_volatile_txn = Some(c.current_tx_id_hash());
+            chain = Some(c);
```

Comment:
Look good just one question

Will this pickup the overlaps between chain 
Persistent data = A -> B
Volatile data = C
Will it be A -> B -> C

Reply:
Yes, we cannot build a chain not from the start, so the volatile cache contains both volatile and persistent parts (or `A -> B -> C` as in the example).

Reply:
Noted, thanks for the calification